js/localization/ldml/date.parser.js

```diff
@@ -179,15 +180,28 @@ var getRegExpInfo = function(format, dateParts) {
             var count = getSameCharCount(format, i),
                 pattern = createPattern(char, count);
 
+            if(stubText) {
+                patterns.push(stubText);
+                regexpText += escapeRegExp(stubText) + ")";
+                stubText = "";
+            }
+
             patterns.push(pattern);
+
             regexpText += "(" + regexpPart(count, dateParts) + ")";
             i += count - 1;
         } else {
-            char = escapeRegExp(char);
-            patterns.push(char);
-            regexpText += "(" + char + ")";
+            if(!stubText) {
+                regexpText += "(";
+            }
+            stubText += char;
         }
     }
+
+    if(stubText) {
+        regexpText += stubText + ")";
```

Comment:
Why is not escapeRegExp(stubText)?

Reply:
Why is not patterns.push(stubText); ?

Reply:
stubText is required without slashes. It should have initial length in the patterns array